src/shared-components/immersiveModal/index.tsx

```diff
@@ -22,193 +23,177 @@ import {
+
+const getHtmlNode = () => document.querySelector('html') || document.body;
+const getDomNode = () =>
+  (document.getElementById(REACT_PORTAL_SECTION_ID) as HTMLElement) ||
+  document.body;
+const getModalMobileScrollingElement = () =>
+  document.getElementById(MODAL_MOBILE_SCROLLING_ID) as HTMLElement;
+const getModalDesktopScrollingElement = () =>
+  document.getElementById(MODAL_DESKTOP_SCROLLING_ID) as HTMLElement;
+
+const ImmersiveModal = ({
+  children,
+  footerContent = null,
+  headerImage = null,
+  onClose,
+  title = '',
+  ...rest
+}: ImmersiveModalProps) => {
+  const [isClosing, setIsClosing] = useState(false);
+  const [showMobileHeaderBar, setShowMobileHeaderBar] = useState(false);
+  const [showDesktopHeaderBar, setShowDesktopHeaderBar] = useState(false);
+
+  const htmlNode = useRef<HTMLElement>(getHtmlNode());
+  const domNode = useRef<HTMLElement>(getDomNode());
+  const modalMobileScrollingElement = useRef<HTMLElement | null>(
+    getModalMobileScrollingElement(),
+  );
+  const modalDesktopScrollingElement = useRef<HTMLElement | null>(
+    getModalDesktopScrollingElement(),
+  );
+
+  const handleScroll = throttle(() => {
+    if (modalMobileScrollingElement.current) {
+      const shouldShowMobileHeaderBar =
+        modalMobileScrollingElement.current.scrollTop > 32;
+      setShowMobileHeaderBar(shouldShowMobileHeaderBar);
+    }
+    if (modalDesktopScrollingElement.current) {
+      const shouldShowDesktopHeaderBar =
+        modalDesktopScrollingElement.current.scrollTop > 32;
+      setShowDesktopHeaderBar(shouldShowDesktopHeaderBar);
+    }
+  }, 100);
 
-  state = {
-    isClosing: false,
-    showMobileHeaderBar: false,
-    showDesktopHeaderBar: false,
+  const handleCloseIntent = () => {
+    setIsClosing(true);
+    setShowMobileHeaderBar(false);
+    setTimeout(onClose, 450);
   };
 
-  htmlNode: HTMLElement;
-
-  domNode: HTMLElement;
-
-  modalMobileScrollingElement: HTMLElement | null = null;
-
-  modalDesktopScrollingElement: HTMLElement | null = null;
-
-  constructor(props: ImmersiveModalProps) {
-    super(props);
-
-    this.htmlNode = document.querySelector('html') || document.body;
-
-    this.domNode =
-      document.querySelector(reactPortalSectionId) || document.body;
-  }
-
-  componentDidMount(): void {
-    this.htmlNode.classList.add('no-scroll');
-
-    this.modalMobileScrollingElement = document.querySelector(
-      '#modal-mobile-scrolling-id',
-    );
-    this.modalDesktopScrollingElement = document.querySelector(
-      '#modal-desktop-scrolling-id',
-    );
-
-    if (this.modalMobileScrollingElement && this.modalDesktopScrollingElement) {
-      this.modalMobileScrollingElement.addEventListener(
-        'scroll',
-        this.handleScroll,
-      );
-      this.modalDesktopScrollingElement.addEventListener(
-        'scroll',
-        this.handleScroll,
-      );
+  const handleEscapeKey = (event: KeyboardEvent) => {
```

Comment:
During our last cycle, I updated the `OffClickWrapper` to treat `Escape` key presses as off-clicks, but I clearly forgot to remove the logic from this component 🙃  We should be able to remove this logic from the `ImmersiveModal` if you like.